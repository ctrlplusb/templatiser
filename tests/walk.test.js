const exec = require('../src')
const walk = require('../src/walk')

describe('walk', () => {
  it('should respect accumulator', () => {
    const tree = exec({
      templatesDir: 'tests/templates',
      inputDir: 'tests/example',
      outputDir: 'tests/output',
      allowedFiles: ['.txt'],
      preprocessSource: x => `preprocessed-${x}`,
    })
    const actual = walk(tree, (acc, { name }) => [...acc, name], [])
    expect(actual).toMatchSnapshot()
  })
})
