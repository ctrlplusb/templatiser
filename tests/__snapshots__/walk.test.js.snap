// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`walk should respect accumulator 1`] = `
Array [
  "cities",
  "cape-town",
  "london",
  "food",
  "fruit",
  "apple",
  "banana",
  "foo",
]
`;
